src/routes/reservations/reservation.svelte

```diff
@@ -0,0 +1,204 @@
+
+    const options = {
+        year: "numeric",
+        month: "numeric",
+        day: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+        second: "numeric",
+        hour12: false,
+    };
+
+    // Toast Settings
+    const toastSuccsess = {
+        message: "Reservierung wurde erfolgreich gelöscht",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-secondary",
+    };
+
+    const toastError = {
+        message: "Reservierung konnte nicht gelöscht werden",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-error",
+    };
+
+    // Policy for restarting backend fetched up to 5 times if there's no reply
+    const retryPolicy = retry(handleAll, {
+        maxAttempts: 5,
+        backoff: new ExponentialBackoff(),
+    });
+
+    // Popup settings
+    const popupClick = {
+        event: "click",
+        target: "popupClick",
+        placement: "top",
+    };
+
+    let carData = new Promise((resolve, reject) => {});
+
+    // Functions
+    function filterDate(dateString) {
+        return new Intl.DateTimeFormat("de-DE", options).format(
+            new Date(dateString),
+        );
+    }
+
+    async function fetchCarForReservation() {
+        carData = new Promise(async (resolve, reject) => {
+            try {
+                const response = await retryPolicy.execute(() =>
+                    fetch(urls.get.singleCar + reservationData.carId, {
+                        credentials: "include",
+                    }),
+                );
+                if (response.ok) {
+                    resolve(await response.json());
+                } else {
+                    throw new Error("No car data found");
+                }
+            } catch (err) {
+                reject(err);
+            }
+        });
+    }
+
+    async function confirmDeletion() {
+        try {
+            const response = await retryPolicy.execute(() =>
+                fetch(
+                    new Request(
+                        urls.delete.singleReservation +
+                            "/" +
+                            reservationData.id,
+                        {
+                            method: "DELETE",
+                            credentials: "include",
+                        },
+                    ),
+                ),
+            );
+            if (response.ok) {
+                location.reload();
+                toastStore.trigger(toastSuccsess);
+            } else {
+                throw new Error("Reservierung konnte nicht gelöscht werden");
+            }
+        } catch (err) {
+            toastStore.trigger(toastError);
+            throw err;
+        }
+    }
+
+    onMount(() => {
+        fetchCarForReservation();
+    });
```

Comment:
```suggestion
    onMount(fetchCarForReservation);
```
Das sollte auch gehen, bitte einmal ausprobieren.

Reply:
Funktioniert, wurde geändert